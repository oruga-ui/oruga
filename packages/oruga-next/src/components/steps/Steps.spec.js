import { mount } from '@vue/test-utils'
import OSteps from '@components/steps/Steps'
import OStepItem from '@components/steps/StepItem'

let wrapper

const WrapperComp = {
    template: `
        <OSteps :value="1">
            <OStepItem></OStepItem>
            <OStepItem :visible="false"></OStepItem>
            <OStepItem :clickable="false"></OStepItem>
        </OSteps>`,
    components: {
        OSteps, OStepItem
    }
}

describe('OSteps', () => {
    beforeEach(() => {
        wrapper = mount(WrapperComp, { sync: false }).find(OSteps)
    })

    it('is called', () => {
        expect(wrapper.exists()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    /*
    it('emit input event with value when active step is modified', async () => {
        wrapper.vm.activeId = wrapper.vm.items[0].value

        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        const valueEmitted = wrapper.emitted().input

        expect(valueEmitted).toBeTruthy()
        expect(valueEmitted.length).toBe(1)
        expect(valueEmitted[0][0]).toBe(0)
    })
    */

    it('manage next/previous listener', async () => {
        // First input
        wrapper.setProps({value: 0})

        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        expect(wrapper.vm.value).toBe(0)

        expect(wrapper.vm.hasNext).toBeTruthy()
        wrapper.vm.next()

        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        // Simulate v-model
        wrapper.setProps(
            {value: wrapper.emitted().input[wrapper.emitted().input.length - 1][0]}
        )

        const emitted = wrapper.emitted().input

        expect(emitted).toBeTruthy()
        expect(emitted.length).toBe(1)
        expect(emitted[0][0]).toBe(3)

        expect(wrapper.vm.hasNext).toBeFalsy()

        wrapper.vm.next()

        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        // Simulate v-model
        wrapper.setProps(
            {value: wrapper.emitted().input[wrapper.emitted().input.length - 1][0]}
        )

        expect(wrapper.emitted().input.length).toBe(1) // To still be

        expect(wrapper.vm.hasPrev).toBeTruthy()
        wrapper.vm.prev()

        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        // Simulate v-model
        wrapper.setProps(
            {value: wrapper.emitted().input[wrapper.emitted().input.length - 1][0]}
        )

        expect(wrapper.emitted().input.length).toBe(2)
        expect(wrapper.emitted().input[1][0]).toBe(1)
        expect(wrapper.vm.hasPrev).toBeFalsy()

        wrapper.vm.prev()
        await wrapper.vm.$nextTick() // Wait until $emits have been handled

        expect(wrapper.emitted().input.length).toBe(2) // To still be
    })

    it('manage wrapper classes as expected', () => {
        expect(wrapper.vm.wrapperClasses[2]['o-steps__wrapper-vertical']).toBeFalsy()

        wrapper.setProps({vertical: true})
        expect(wrapper.vm.wrapperClasses[2]['o-steps__wrapper-vertical']).toBeTruthy()

        wrapper.setProps({position: 'right'})
        expect(wrapper.vm.wrapperClasses[3]['o-steps__wrapper-position-right']).toBeTruthy()
    })

    it('throws an error when there is no item', () => {
        const spy = jest.spyOn(global.console, 'error').mockImplementation(() => {})

        try {
            wrapper = mount({
                template: `<OSteps/>`,
                components: {
                    OSteps
                },
                destroyed() {
                    spy()
                }
            })
        } catch (error) {
            expect(error.message).stringMatching(/item inside/)
        } finally {
            spy.mockRestore()
        }
    })
})
