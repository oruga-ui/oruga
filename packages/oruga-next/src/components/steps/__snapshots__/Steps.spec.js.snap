// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSteps render correctly 1`] = `
<div class="o-steps__wrapper">
    <nav class="o-steps o-steps--animated">
        <div class="o-steps__nav-item o-steps__nav-item-active">
            <!----> <a class="o-steps__link o-steps__link-label-bottom">
                <div class="o-steps__marker o-steps__marker--rounded">
                    <!---->
                </div>
                <div class="o-steps__title"></div>
            </a>
        </div>
        <div class="o-steps__nav-item" style="display: none;"><span class="o-steps__divider"></span> <a class="o-steps__link o-steps__link-label-bottom">
                <div class="o-steps__marker o-steps__marker--rounded">
                    <!---->
                </div>
                <div class="o-steps__title"></div>
            </a></div>
        <div class="o-steps__nav-item"><span class="o-steps__divider"></span> <a class="o-steps__link o-steps__link-label-bottom">
                <div class="o-steps__marker o-steps__marker--rounded">
                    <!---->
                </div>
                <div class="o-steps__title"></div>
            </a></div>
    </nav>
    <section class="o-steps__content o-steps__content-transitioning">
        <div class="o-steps__item v-enter v-enter-active" style=""></div>
        <div class="o-steps__item" style="display: none;"></div>
        <div class="o-steps__item" style="display: none;"></div>
    </section>
    <nav class="o-steps__navigation"><button type="button" role="button" class="o-btn o-btn--disabled"><span class="o-btn__wrapper"><span class="o-icon o-btn__icon" style="transform: rotate(0deg);"><i class="mdi mdi-chevron-left mdi-24px"></i></span>
            <!---->
            <!----></span>
        </button> <button type="button" role="button" class="o-btn"><span class="o-btn__wrapper"><span class="o-icon o-btn__icon" style="transform: rotate(0deg);"><i class="mdi mdi-chevron-right mdi-24px"></i></span>
            <!---->
            <!----></span>
        </button></nav>
</div>
`;
